/* -------------------------------------------------------------------------- */
/* Copyright 2002-2025, OpenNebula Project, OpenNebula Systems                */
/*                                                                            */
/* Licensed under the Apache License, Version 2.0 (the "License"); you may    */
/* not use this file except in compliance with the License. You may obtain    */
/* a copy of the License at                                                   */
/*                                                                            */
/* http://www.apache.org/licenses/LICENSE-2.0                                 */
/*                                                                            */
/* Unless required by applicable law or agreed to in writing, software        */
/* distributed under the License is distributed on an "AS IS" BASIS,          */
/* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   */
/* See the License for the specific language governing permissions and        */
/* limitations under the License.                                             */
/* -------------------------------------------------------------------------- */
syntax = "proto3";

import "shared.proto";

package one.group;

option go_package = "github.com/OpenNebula/one/src/oca/go/src/goca/api/group";

message AllocateRequest
{
  string session_id = 1;
  string gname      = 2;
}

message DeleteRequest
{
  string session_id = 1;
  int32 oid         = 2;
}

message InfoRequest
{
  string session_id = 1;
  int32 oid         = 2;
  bool decrypt      = 3;
}

message UpdateRequest
{
  string session_id = 1;
  int32 oid         = 2;
  string template   = 3;
  int32 append      = 4;
}

message AddAdminRequest
{
  string session_id = 1;
  int32 oid         = 2;
  int32 user_id     = 3;
}

message DelAdminRequest
{
  string session_id = 1;
  int32 oid         = 2;
  int32 user_id     = 3;
}

message QuotaRequest
{
  string session_id = 1;
  int32 oid         = 2;
  string quota      = 3;
}

message DefaultQuotaUpdateRequest
{
  string session_id = 1;
  string quota      = 2;
}

message PoolInfoRequest
{
  string session_id = 1;
}

service GroupService
{
  rpc Allocate (one.group.AllocateRequest) returns (one.ResponseID);

  rpc Delete (one.group.DeleteRequest) returns (one.ResponseID);

  rpc Info (one.group.InfoRequest) returns (one.ResponseXML);

  rpc Update (one.group.UpdateRequest) returns (one.ResponseID);

  rpc Quota (one.group.QuotaRequest) returns (one.ResponseID);

  rpc AddAdmin (one.group.AddAdminRequest) returns (one.ResponseID);

  rpc DelAdmin (one.group.DelAdminRequest) returns (one.ResponseID);

  rpc DefaultQuotaInfo (one.group.PoolInfoRequest) returns (one.ResponseXML);

  rpc DefaultQuotaUpdate (one.group.DefaultQuotaUpdateRequest) returns (one.ResponseXML);

  rpc PoolInfo (one.group.PoolInfoRequest) returns (one.ResponseXML);
}
